### PR TITLE
Implementation of boss

### PR DESCRIPTION
**Pull request by Ragnarok77-factorio**
### Brief description of the changes:
Goal was to implement boss biters while trying to keep as much as possible current balance.
Since we are limited on units change, a special force _boss was added for biters to be able to edit their attack speed and damage without changing current biters stats (boss biters have their own stats). After this update, biters are a lot less heavy on factorio because there are less biters (but the current wave strength is the same as before)

Right now I let the useless comment and functions to be able for anyone to test boss easily (spawn a wave for example, debug message), but once the code is reviewed, I will delete all the test functions/debug messages.

I let this as draft until test functions/debug messages are removed, but please check it and comment !

Result of vote for feature  : https://discord.com/channels/823696400797138974/823771211421974579/972336162041655386

Please read the thread in dev-git before looking the code, to understand more the changes behind it : https://discord.com/channels/823696400797138974/979366629899137054

To test balance & ups/fps : just do /c global.wave1 =1 true or wave2 or wave3 or wave4 or even change the values of the waves if you want (more biters, less biters..and so on)

### Tested Changes:
- [x] I've tested the changes locally or with people.
- [ ] I've not tested the changes.

### CODE REVIEW

**Comment by Ragnarok77-factorio** [comment]
Okay, cleaned all debug messages and spawn wave system method and so on, I think it will be too chaotic to get code reviewed, then cleaned, then reviewed again... If needed, you can always checkout the commit before cleaning : [099d3ae]..etc

File: maps/biter_battles_v2/feeding.lua [hunk 1]
```diff
@@ -19,6 +19,21 @@ function get_instant_threat_player_count_modifier()
 	return m
 end
 
+
+local function update_boss_modifiers(force_name_biter,damage_mod_mult,speed_mod_mult)
+	local damage_mod = math.round((global.bb_evolution[force_name_biter]) * 1.0, 3) * damage_mod_mult
```
Comment (TheBigZet) [inline maps/biter_battles_v2/feeding.lua:24]: use the magical `math_round` here? It feels like it will be called frequently

**Review by TheBigZet — CHANGES_REQUESTED** [review]
Add somewhere `require "utils.profiler` so we could meastre real life performance.
Also maybe flying text above the island saying this is experimental version?

File: maps/biter_battles_v2/feeding.lua (hunk 1 above)
Comment (Ragnarok77-factorio) [reply maps/biter_battles_v2/feeding.lua]: Indeed, good call, I updated for math_floor too.

**Comment by Ragnarok77-factorio** [comment]
> Add somewhere `require "utils.profiler` so we could meastre real life performance. Also maybe flying text above the island saying this is experimental version?

Understood, added a flying text and utils profiler, not sure where you wanted it though. @TheBigZet

**Comment by TheBigZet** [comment]
Anywhere would be good, we should be able to start a profiler from a command line this way (we can only require modules in runtime that were required on init(

**Comment by Ragnarok77-factorio** [comment]
> Anywhere would be good, we should be able to start a profiler from a command line this way (we can only require modules in runtime that were required on init(

Added it, should be fine I guess..?

File: maps/biter_battles_v2/init.lua [hunk 2]
```diff
@@ -160,6 +160,7 @@ function Public.draw_structures()
 	Terrain.generate_additional_rocks(surface)
 	Terrain.generate_silo(surface)
 	Terrain.draw_spawn_circle(surface)
+	game.surfaces[global.bb_surface_name].create_entity({name="flying-text",position={-4,-13},text="Experimental version",color = {r = 0, g = 1, b = 0, a = 0.5},text_align="right"}).active=false
```
Comment (TheBigZet) [inline maps/biter_battles_v2/init.lua:163]: `Experimental version: Boss biters`

**Review by TheBigZet — CHANGES_REQUESTED** [review]

File: maps/biter_battles_v2/init.lua (hunk 2 above)
Comment (Ragnarok77-factorio) [reply maps/biter_battles_v2/init.lua]: Added it @TheBigZet

**Pull request merged**